### PR TITLE
First Commit

### PR DESCRIPTION
**Pull request by hamimraavi**
Sample URL:
http://127.0.0.1:8000/moviesearch/?t=moviename
Example:
http://127.0.0.1:8000/moviesearch/?t=inception

Works for spaces within movie names, incorrect movie names, and movies yet to be released.

TODO:
Return multiple movies that match the given search string

### CODE REVIEW

File: moviesearch/search.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
+from BeautifulSoup import BeautifulSoup
+from mechanize import Browser
+import re
+import pdb
```
Comment (vishesh92) [inline moviesearch/search.py:4]: rearrange your imports.
https://www.python.org/dev/peps/pep-0008/#imports

Comment (vishesh92) [inline moviesearch/search.py:4]: pdb in code?

File: moviesearch/search.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+import json
+
+def get_soup(movie): 
+    movie = '+'.join(movie.split())
+    url = "http://www.imdb.com/find?ref_=nv_sr_fn&q="+movie+"&s=all"
```
Comment (vishesh92) [inline moviesearch/search.py:9]: Space around operators.

File: myimdb/urls.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    2. Add a URL to urlpatterns:  url(r'^$', Home.as_view(), name='home')
+Including another URLconf
+    1. Import the include() function: from django.conf.urls import url, include
+    2. Add a URL to urlpatterns:  url(r'^blog/', include('blog.urls'))
+"""
+from django.conf.urls import include, url
+from django.contrib import admin
+
+urlpatterns = [
+    #url(r'^moviesearch/([a-z]+)/', include('moviesearch.urls')),
```
Comment (vishesh92) [inline myimdb/urls.py:20]: Remove commented code.

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,57 @@
+altgraph==0.10.2
```
Comment (vishesh92) [inline requirements.txt:1]: Do you really need all these modules?

File: moviesearch/search.py (hunk 2 above)
Comment (vaidik) [inline moviesearch/search.py:9]: Use string formatting instead of concatenation. https://pyformat.info/

File: moviesearch/search.py [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+    br = Browser()
+    br.open(url)
+    try:
+        link = list(br.links(url_regex = re.compile(r"/title/tt*")))[0]
+        print link
```
Comment (vaidik) [inline moviesearch/search.py:14]: Why print?

File: moviesearch/search.py [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+    except:
+        return ""
```
Comment (vaidik) [inline moviesearch/search.py:16]: What's the point of this return? How does this help?

File: moviesearch/views.py [hunk 7]
```diff
@@ -0,0 +1,14 @@
+from django.http import HttpResponse
+import search
+import pdb
+
+def index(request):
+    res = search.get_all_details("inception")
+    return HttpResponse(res)
+
+def results1(request):
```
Comment (vaidik) [inline moviesearch/views.py:9]: `results1`?

File: moviesearch/search.py [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+    except:
+        return "N/A"
+
+def get_all_details(movie):
+    response = []
+
+    soup = get_soup(movie)
+    
+    title = get_title(movie, soup)
+    if title == "N/A":
```
Comment (vaidik) [inline moviesearch/search.py:119]: Might as well just use `None`. String comparison would be relatively expensive.

File: moviesearch/search.py [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+    response.append({'Year': year})
+    response.append({'IMDb Rating': rating})
+    response.append({'Votes': votes})
+    response.append({'Runtime': duration})
+    response.append({'Rated': content})
+    response.append({'Genre': genre})
+    response.append({'Director': director})
+    response.append({'Writer': writer})
+    response.append({'Actors': actors})
+    return json.dumps(response)
```
Comment (vaidik) [inline moviesearch/search.py:146]: Can you share the response here? Exactly what is it that you are returning?

Comment (hamimraavi) [inline moviesearch/search.py:146]: [{"Title": "Gladiator (2000) - IMDb"}, {"Year": "2000"}, {"IMDb Rating": "8.5"}, {"Votes": "967,129"}, {"Runtime": "2h 35min"}, {"Rated": "Rated R for intense, graphic combat"}, {"Genre": ["Action", "Drama"]}, {"Director": ["Ridley Scott"]}, {"Writer": ["David Franzoni", "David Franzoni", "DreamWorks SKG", "Universal Pictures", "Scott Free Productions"]}, {"Actors": ["Russell Crowe", "Joaquin Phoenix", "Connie Nielsen"]}]

File: moviesearch/search.py (hunk 5 above)
Comment (hamimraavi) [inline moviesearch/search.py:14]: fixed

File: requirements.txt (hunk 4 above)
Comment (hamimraavi) [inline requirements.txt:1]: fixed

**Pull request closed without merging**